back/app/models/concerns/user_roles.rb

```diff
@@ -116,18 +116,33 @@ def admin?
     roles.any? { |r| r['type'] == 'admin' }
   end
 
-  # Returns roles excluding the `project_moderator` roles that are redundant with
-  # `project_folder_moderator` roles (i.e. the user is a project moderator for a
-  # project that is in a folder that they moderate).
-  def compacted_roles
-    redundant_project_ids = AdminPublication
-      .joins(:parent)
-      .where(parent: { publication_id: moderated_project_folder_ids })
-      .pluck(:publication_id)
-
-    roles.reject do |role|
-      role['type'] == 'project_moderator' && role['project_id'].in?(redundant_project_ids)
+  # Reduce the roles to add to the JWT cookie to stop it getting too large
+  # Roles from cookie are ONLY used by workshops and cl2-admin-templates
+  # - workshops only cares if it finds type: admin or type: project_moderator.
+  #   It does not use project_id or use type: project_moderator
+  # - cl2-admin-templates only cares if it finds type: admin or type: project_folder_moderator.
+  #   It needs the project_folder_id but does not use type: project_moderator
+  def compress_roles
+    admin = false
+    project_moderator = false
+    compressed_roles = []
+
+    roles.each do |role|
+      case role['type']
+      when 'admin'
+        admin = true
+      when 'project_moderator'
+        project_moderator = true
+      when 'project_folder_moderator'
+        project_moderator = true
```

Comment:
Added this additionally as workshops only looks for `type: project_moderator` so currently if a user is only a folder moderator, workshops will not see them as admin.